pkgs/development/compilers/crystal/default.nix

```diff
@@ -64,12 +67,11 @@ let
         patchShebangs $out/bin/crystal
       '';
 
-      meta.broken = lib.versionOlder version "1.2.0" && isAarch64Darwin;
+      meta.broken = (lib.versionOlder version "1.2.0" && isAarch64Darwin) || (lib.versionAtLeast version "1.3.0" && stdenv.system == "i686-linux") || (lib.versionAtLeast version "1.5.0" && stdenv.system == "aarch64-linux");
```

Comment:
> ofborg-eval — This PR does not cleanly list package outputs after merging. 

Would it make sense to set `meta.platforms` of the binary derivations based on the keys of `sha256s`, and let the `generic` derivation inherit `binary.meta.platforms`? If there's no upstream release for a platform, then it seems fine to exclude that from the list of supported platforms, rather than saying that the package is broken.

I made these changes locally:

```diff
diff --git a/pkgs/development/compilers/crystal/default.nix b/pkgs/development/compilers/crystal/default.nix
index 67b9296b..cb3ff680 100644
--- a/pkgs/development/compilers/crystal/default.nix
+++ b/pkgs/development/compilers/crystal/default.nix
@@ -67,7 +67,7 @@ let
         patchShebangs $out/bin/crystal
       '';
 
-      meta.broken = (lib.versionOlder version "1.2.0" && isAarch64Darwin) || (lib.versionAtLeast version "1.3.0" && stdenv.system == "i686-linux") || (lib.versionAtLeast version "1.5.0" && stdenv.system == "aarch64-linux");
+      meta.platforms = lib.attrNames sha256s;
     };
 
   commonBuildInputs = extraBuildInputs: [
@@ -237,13 +237,11 @@ let
       };
 
       meta = with lib; {
-        broken = stdenv.isDarwin;
+        inherit (binary.meta) platforms;
         description = "A compiled language with Ruby like syntax and type inference";
         homepage = "https://crystal-lang.org/";
         license = licenses.asl20;
         maintainers = with maintainers; [ david50407 manveru peterhoeg ];
-        platforms = let archNames = builtins.attrNames archs; in
-          if (lib.versionOlder version "1.2.0") then remove "aarch64-darwin" archNames else archNames;
       };
     })
   );
```

and ran (what I think are) the [steps for reproducing ofborg-eval's test](https://github.com/NixOS/ofborg#running-meta-checks-locally):

```sh
curl -o outpaths.nix https://raw.githubusercontent.com/NixOS/ofborg/released/ofborg/src/outpaths.nix
GC_INITIAL_HEAP_SIZE=4g nix-env -f ./outpaths.nix -qaP --no-name --out-path --arg checkMeta true --show-trace > out-paths
```

Without the changes above, I saw the same error as the one in the [gist that has ofborg-eval's check's details](https://gist.github.com/GrahamcOfBorg/d87d2ff07b17141df6f73f32dfe16f75), "attribute 'aarch64-linux' missing", and with the changes, the command ran long enough that it must've gotten past that part. (I had to terminate it after my machine became near-unresponsive for more than 30 minutes.)

Reply:
Applied and can confirm it works locally, thanks!

I'll leave this as unresolved, in-case one of the maintainers has any input as I'm not very familiar with nix.